File: Dot examples/ASP Book Examples/Book Examples/Chapter27/CS/App_Code/CustomBuildProviders/SimpleBuildProvider.cs
using System;
using System.Web.Compilation;
using System.CodeDom;
using System.IO;

namespace AspNetUnleashed
{
    public class SimpleBuildProvider : BuildProvider
    {
        public override void GenerateCode(AssemblyBuilder ab)
        {
            string fileName = Path.GetFileNameWithoutExtension(this.VirtualPath);
            string snippet = "public class " + fileName + @"
                {
                    public static void DoSomething(){} 
                }";
            ab.AddCodeCompileUnit(this, new CodeSnippetCompileUnit(snippet));
        }

    }
}

File: Dot examples/ASP Book Examples/Book Examples/Chapter27/VB/App_Code/CustomBuildProviders/SimpleBuildProvider.vb
Imports System
Imports System.Web.Compilation
Imports System.CodeDom
Imports System.IO

Namespace AspNetUnleashed
    Public Class SimpleBuildProvider
        Inherits BuildProvider

        Public Overrides Sub GenerateCode(ByVal ab As AssemblyBuilder)
            Dim fileName As String = Path.GetFileNameWithoutExtension(Me.VirtualPath)
            Dim snippet As String = "Public Class " & fileName & vbNewLine
            snippet &= "  Public Shared Sub DoSomething()" & vbNewLine
            snippet &= "  End Sub" & vbNewLine
            snippet &= "End Class"
            ab.AddCodeCompileUnit(Me, New CodeSnippetCompileUnit(snippet))
        End Sub

    End Class
End Namespace

